src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -0,0 +1,337 @@
+                    releaseList.Add(item);
+
+                }
+                catch (Exception e)
+                {
+                    Logger.LogError($"Some problem happened while starting publishing pipeline " +
+                        $"{pipeline.ReleasePipeline.PipelineIdentifier} for build " +
+                        $"{build.AzureDevOpsBuildId}: {e.Message}", e);
+                    throw;
+                }
+            }
+
+            if (releaseList.Count > 0)
+            {
+                using (ITransaction tx = StateManager.CreateTransaction())
+                {
+                    var runningPipelinesForBuild = await runningPipelines.TryGetValueAsync(tx, buildId);
+                    if (runningPipelinesForBuild.HasValue)
+                    {
+                        // Some channel already triggered release pipelines for this build. Need to update with the releases for the new channel.
+                        releaseList.AddRange(runningPipelinesForBuild.Value);
+                        await runningPipelines.AddOrUpdateAsync(tx, buildId, releaseList, (key, oldValue) => releaseList);
+                    }
+                    else
+                    {
+                        await runningPipelines.AddAsync(tx, buildId, releaseList);
+                    }
+                    await tx.CommitAsync();
+                }
+            }
+            await ProcessFinishedReleasesAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Check for finished releases and add the corresponding build channel association to the DB.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        // Check every 10 minutes for any finished release Pipelines
+        [CronSchedule("0 0/10 * 1/1 * ? *", TimeZones.UTC)]
+        public async Task ProcessFinishedReleasesAsync(CancellationToken cancellationToken)
+        {
+            var runningPipelines =
+                        await StateManager.GetOrAddAsync<IReliableDictionary<int, IList<ReleasePipelineStatusItem>>>(RunningPipelineDictionaryName);
+            try
+            {
+                HashSet<BuildChannel> buildChannelsToAdd = new HashSet<BuildChannel>(new BuildChannelComparer());
+                using (ITransaction tx = StateManager.CreateTransaction())
+                {
+                    var runningPipelinesEnumerable = await runningPipelines.CreateEnumerableAsync(tx, EnumerationMode.Unordered);
+                    var asyncEnumerator = runningPipelinesEnumerable.GetAsyncEnumerator();
+                    while (await asyncEnumerator.MoveNextAsync(cancellationToken))
+                    {
+                        int buildId = asyncEnumerator.Current.Key;
+                        ConditionalValue<IList<ReleasePipelineStatusItem>> maybeReleaseStatuses = await runningPipelines.TryGetValueAsync(tx, buildId);
```

Comment:
This isn't needed. You have the value already in `asyncEnumerator.Current.Value`

Reply:
Yep. That was a derp while just trying to learn about the reliable collections. fixed now